controllers/grid/queries/form/QueryNoteForm.inc.php

```diff
@@ -141,24 +145,39 @@ public function execute(...$functionArgs)
         $query = $this->getQuery();
         if ($query->getIsClosed()) {
             $headNote = $query->getHeadNote();
-            if ($user->getId() != $headNote->getUserId()) {
+            if ($currentUser->getId() != $headNote->getUserId()) {
                 // Re-open the query.
                 $query->setIsClosed(false);
                 $queryDao = DAORegistry::getDAO('QueryDAO'); /** @var QueryDAO $queryDao */
                 $queryDao->updateObject($query);
             }
         }
 
+        $submissionId = $query->getAssocId();
+        $submission = Repo::submission()->get($submissionId);
+
+        $mailable = new MailDiscussionMessage($request->getContext(), $submission, $query);
+        $emailTemplate = Repo::emailTemplate()->getByKey($request->getContext()->getId(), $mailable::getEmailTemplateKey());
+        $context = $request->getContext();
```

Comment:
Throughout this function you mix use of `$request->getContext()` and `$context`. I don't mind whichever you prefer, but I'd recommend keeping it consistent throughout the function.